// @multiversx/sdk-core/out/smartcontracts/codec/explicit-enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExplicitEnumBinaryCodec = void 0;
const typesystem_1 = require("../typesystem");
const explicit_enum_1 = require("../typesystem/explicit-enum");
const string_1 = require("./string");
class ExplicitEnumBinaryCodec {
    constructor() {
        this.stringCodec = new string_1.StringBinaryCodec();
    }
    decodeTopLevel(buffer, type) {
        const stringValue = this.stringCodec.decodeTopLevel(buffer);
        return new explicit_enum_1.ExplicitEnumValue(type, new explicit_enum_1.ExplicitEnumVariantDefinition(stringValue.valueOf()));
    }
    decodeNested(buffer, type) {
        const [value, length] = this.stringCodec.decodeNested(buffer);
        const enumValue = new explicit_enum_1.ExplicitEnumValue(type, new explicit_enum_1.ExplicitEnumVariantDefinition(value.valueOf()));
        return [enumValue, length];
    }
    encodeNested(enumValue) {
        const buffer = this.stringCodec.encodeNested(new typesystem_1.StringValue(enumValue.valueOf().name));
        return buffer;
    }
    encodeTopLevel(enumValue) {
        const buffer = this.stringCodec.encodeTopLevel(new typesystem_1.StringValue(enumValue.valueOf().name));
        return buffer;
    }
}
exports.ExplicitEnumBinaryCodec = ExplicitEnumBinaryCodec;
