// @multiversx/sdk-web-wallet-iframe-provider/out/IframeManager/IframeManager.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toggleIconElementStyle = exports.titleElementStyle = exports.headingElementStyle = exports.iframeStyle = exports.bodyStyle = exports.headerStyle = exports.containerStyle = void 0;
exports.containerStyle = `
    z-index: 9999;
    position: fixed;
    visibility: hidden;
    right: 0;
    left: auto;
    border: none;
    border-radius: 12px;
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
    height: calc(100vh - 8px - 64px);
    transition: all 400ms ease;
    width: 420px;
    max-width: calc(100vw - 8px);
    background: black;
    overflow: hidden;
    top: 64px;
  `;
exports.headerStyle = `
    color: #fff;
    cursor: pointer;
    height: 80px;
    border-radius: 12px;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `;
exports.bodyStyle = `
    height: calc(100vh - 80px - 64px - 8px);
    width: 420px;
    max-width: calc(100vw - 8px);
    overflow: hidden;
  `;
exports.iframeStyle = `
    width: 100%;
    height: 100%;
    border: none;
  `;
exports.headingElementStyle = `
  display: flex;
  align-items: center;
  cursor: pointer;
  gap: 12px;
  width: 100%;
  position: relative;
  padding: 0 24px;
  height: 80px;
`;
exports.titleElementStyle = `
  font-size: 20px;
  font-family: 'Roobert Medium', sans-serif;
  color: #E5E5E5;
  transition: all 400ms ease;
  line-height: 1;
  user-select: none;
`;
exports.toggleIconElementStyle = `
  margin-left: auto;
  transition: all 400ms ease;
`;
