// @multiversx/sdk-core/out/networkProviders/transactionEvents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionEventTopic = exports.TransactionEventData = exports.TransactionEvent = void 0;
const address_1 = require("../address");
class TransactionEvent {
    constructor(init) {
        this.address = address_1.Address.empty();
        this.identifier = "";
        this.topics = [];
        /**
         * @deprecated Use "dataPayload" instead.
         */
        this.data = "";
        this.dataPayload = new TransactionEventData(Buffer.from("", "utf8"));
        this.additionalData = [];
        Object.assign(this, init);
    }
    static fromHttpResponse(responsePart) {
        let result = new TransactionEvent();
        result.address = new address_1.Address(responsePart.address);
        result.identifier = responsePart.identifier || "";
        result.topics = (responsePart.topics || []).map((topic) => new TransactionEventTopic(topic));
        result.dataPayload = TransactionEventData.fromBase64(responsePart.data);
        result.additionalData = (responsePart.additionalData || []).map(TransactionEventData.fromBase64);
        result.data = result.dataPayload.toString();
        return result;
    }
    findFirstOrNoneTopic(predicate) {
        return this.topics.filter((topic) => predicate(topic))[0];
    }
    getLastTopic() {
        return this.topics[this.topics.length - 1];
    }
}
exports.TransactionEvent = TransactionEvent;
class TransactionEventData {
    constructor(data) {
        this.raw = data;
    }
    static fromBase64(str) {
        return new TransactionEventData(Buffer.from(str || "", "base64"));
    }
    toString() {
        return this.raw.toString("utf8");
    }
    hex() {
        return this.raw.toString("hex");
    }
    valueOf() {
        return this.raw;
    }
}
exports.TransactionEventData = TransactionEventData;
class TransactionEventTopic {
    constructor(topic) {
        this.raw = Buffer.from(topic || "", "base64");
    }
    toString() {
        return this.raw.toString("utf8");
    }
    hex() {
        return this.raw.toString("hex");
    }
    valueOf() {
        return this.raw;
    }
}
exports.TransactionEventTopic = TransactionEventTopic;
