// @multiversx/sdk-core/out/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ESDT_CONTRACT_ADDRESS = exports.DELEGATION_MANAGER_SC_ADDRESS = exports.CONTRACT_DEPLOY_ADDRESS = exports.BECH32_ADDRESS_LENGTH = exports.DEFAULT_HRP = exports.EGLD_IDENTIFIER_FOR_MULTI_ESDTNFT_TRANSFER = exports.UNKNOWN_SIGNER = exports.SDK_JS_SIGNER = exports.METACHAIN_ID = exports.WasmVirtualMachine = exports.CURRENT_NUMBER_OF_SHARDS_WITHOUT_META = exports.HEX_TRANSACTION_HASH_LENGTH = exports.MESSAGE_PREFIX = exports.DEFAULT_MESSAGE_VERSION = exports.ESDT_CONTRACT_ADDRESS_HEX = exports.DELEGATION_MANAGER_SC_ADDRESS_HEX = exports.CONTRACT_DEPLOY_ADDRESS_HEX = exports.VM_TYPE_WASM_VM = exports.ARGUMENTS_SEPARATOR = exports.ESDT_TRANSFER_VALUE = exports.MULTI_ESDTNFT_TRANSFER_FUNCTION_NAME = exports.ESDTNFT_TRANSFER_FUNCTION_NAME = exports.ESDT_TRANSFER_FUNCTION_NAME = exports.ESDT_TRANSFER_GAS_LIMIT = exports.MIN_TRANSACTION_VERSION_THAT_SUPPORTS_OPTIONS = exports.TRANSACTION_VERSION_DEFAULT = exports.TRANSACTION_OPTIONS_TX_GUARDED = exports.TRANSACTION_OPTIONS_TX_HASH_SIGN = exports.TRANSACTION_OPTIONS_DEFAULT = exports.TRANSACTION_MIN_GAS_PRICE = void 0;
exports.TRANSACTION_MIN_GAS_PRICE = 1000000000;
exports.TRANSACTION_OPTIONS_DEFAULT = 0;
exports.TRANSACTION_OPTIONS_TX_HASH_SIGN = 0b0001;
exports.TRANSACTION_OPTIONS_TX_GUARDED = 0b0010;
exports.TRANSACTION_VERSION_DEFAULT = 2;
exports.MIN_TRANSACTION_VERSION_THAT_SUPPORTS_OPTIONS = 2;
exports.ESDT_TRANSFER_GAS_LIMIT = 500000;
exports.ESDT_TRANSFER_FUNCTION_NAME = "ESDTTransfer";
exports.ESDTNFT_TRANSFER_FUNCTION_NAME = "ESDTNFTTransfer";
exports.MULTI_ESDTNFT_TRANSFER_FUNCTION_NAME = "MultiESDTNFTTransfer";
exports.ESDT_TRANSFER_VALUE = "0";
exports.ARGUMENTS_SEPARATOR = "@";
exports.VM_TYPE_WASM_VM = new Uint8Array([0x05, 0x00]);
exports.CONTRACT_DEPLOY_ADDRESS_HEX = "0000000000000000000000000000000000000000000000000000000000000000";
exports.DELEGATION_MANAGER_SC_ADDRESS_HEX = "000000000000000000010000000000000000000000000000000000000004ffff";
exports.ESDT_CONTRACT_ADDRESS_HEX = "000000000000000000010000000000000000000000000000000000000002ffff";
exports.DEFAULT_MESSAGE_VERSION = 1;
exports.MESSAGE_PREFIX = "\x17Elrond Signed Message:\n";
exports.HEX_TRANSACTION_HASH_LENGTH = 64;
exports.CURRENT_NUMBER_OF_SHARDS_WITHOUT_META = 3;
exports.WasmVirtualMachine = "0500";
exports.METACHAIN_ID = 4294967295;
exports.SDK_JS_SIGNER = "sdk-js";
exports.UNKNOWN_SIGNER = "unknown";
exports.EGLD_IDENTIFIER_FOR_MULTI_ESDTNFT_TRANSFER = "EGLD-000000";
/**
 * @deprecated
 */
exports.DEFAULT_HRP = "erd";
/**
 * @deprecated
 */
exports.BECH32_ADDRESS_LENGTH = 62;
/**
 * @deprecated Use {@link CONTRACT_DEPLOY_ADDRESS_HEX} instead.
 */
exports.CONTRACT_DEPLOY_ADDRESS = "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu";
/**
 * @deprecated Use {@link DELEGATION_MANAGER_SC_ADDRESS_HEX} instead.
 */
exports.DELEGATION_MANAGER_SC_ADDRESS = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6";
/**
 * @deprecated Use {@link 000000000000000000010000000000000000000000000000000000000002ffff} instead.
 */
exports.ESDT_CONTRACT_ADDRESS = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u";
