// @multiversx/sdk-core/out/signableMessage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignableMessage = void 0;
const address_1 = require("./address");
const signature_1 = require("./signature");
const constants_1 = require("./constants");
const createKeccakHash = require("keccak");
/**
 * @deprecated Use {@link Message} instead.
 */
class SignableMessage {
    constructor(init) {
        this.message = Buffer.from([]);
        this.signature = Buffer.from([]);
        this.version = 1;
        this.signer = "ErdJS";
        this.address = address_1.Address.empty();
        Object.assign(this, init);
    }
    serializeForSigning() {
        const messageSize = Buffer.from(this.message.length.toString());
        const signableMessage = Buffer.concat([messageSize, this.message]);
        let bytesToHash = Buffer.concat([Buffer.from(constants_1.MESSAGE_PREFIX), signableMessage]);
        return createKeccakHash("keccak256").update(bytesToHash).digest();
    }
    serializeForSigningRaw() {
        return Buffer.concat([this.getMessageSize(), this.message]);
    }
    getSignature() {
        return this.signature;
    }
    applySignature(signature) {
        this.signature = signature_1.interpretSignatureAsBuffer(signature);
    }
    getMessageSize() {
        const messageSize = Buffer.alloc(4);
        messageSize.writeUInt32BE(this.message.length, 0);
        return messageSize;
    }
    toJSON() {
        return {
            address: this.address.bech32(),
            message: "0x" + this.message.toString("hex"),
            signature: "0x" + this.signature.toString("hex"),
            version: this.version,
            signer: this.signer,
        };
    }
}
exports.SignableMessage = SignableMessage;
