// @multiversx/sdk-core/out/wallet/mnemonic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mnemonic = void 0;
const ed25519_hd_key_1 = require("ed25519-hd-key");
const errors_1 = require("../errors");
const userKeys_1 = require("./userKeys");
const MNEMONIC_STRENGTH = 256;
const BIP44_DERIVATION_PREFIX = "m/44'/508'/0'/0'";
let bip39;
// Load bip39 when needed
function loadBip39() {
    if (!bip39) {
        try {
            bip39 = require("bip39");
        }
        catch (error) {
            throw new Error("bip39 is required but not installed. Please install 'bip39' to use mnemonic features.");
        }
    }
}
class Mnemonic {
    constructor(text) {
        this.text = text;
    }
    static generate() {
        loadBip39();
        const text = bip39.generateMnemonic(MNEMONIC_STRENGTH);
        return new Mnemonic(text);
    }
    static fromString(text) {
        loadBip39();
        text = text.trim();
        Mnemonic.assertTextIsValid(text);
        return new Mnemonic(text);
    }
    static fromEntropy(entropy) {
        loadBip39();
        try {
            const text = bip39.entropyToMnemonic(Buffer.from(entropy));
            return new Mnemonic(text);
        }
        catch (err) {
            throw new errors_1.ErrBadMnemonicEntropy(err);
        }
    }
    static assertTextIsValid(text) {
        loadBip39();
        let isValid = bip39.validateMnemonic(text);
        if (!isValid) {
            throw new errors_1.ErrWrongMnemonic();
        }
    }
    deriveKey(addressIndex = 0, password = "") {
        loadBip39();
        let seed = bip39.mnemonicToSeedSync(this.text, password);
        let derivationPath = `${BIP44_DERIVATION_PREFIX}/${addressIndex}'`;
        let derivationResult = ed25519_hd_key_1.derivePath(derivationPath, seed.toString("hex"));
        let key = derivationResult.key;
        return new userKeys_1.UserSecretKey(key);
    }
    getWords() {
        return this.text.split(" ");
    }
    getEntropy() {
        loadBip39();
        const entropy = bip39.mnemonicToEntropy(this.text);
        return Buffer.from(entropy, "hex");
    }
    toString() {
        return this.text;
    }
}
exports.Mnemonic = Mnemonic;
