// @multiversx/sdk-core/out/wallet/validatorKeys.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidatorPublicKey = exports.ValidatorSecretKey = exports.BLS = exports.VALIDATOR_PUBKEY_LENGTH = exports.VALIDATOR_SECRETKEY_LENGTH = void 0;
const errors_1 = require("../errors");
const assertions_1 = require("./assertions");
const pem_1 = require("./pem");
exports.VALIDATOR_SECRETKEY_LENGTH = 32;
exports.VALIDATOR_PUBKEY_LENGTH = 96;
class BLS {
    static loadBLSModule() {
        if (!BLS.bls) {
            try {
                BLS.bls = require("@multiversx/sdk-bls-wasm");
            }
            catch (error) {
                throw new Error("BLS module is required but not installed. Please install '@multiversx/sdk-bls-wasm'.");
            }
        }
    }
    static async initIfNecessary() {
        if (BLS.isInitialized) {
            return;
        }
        BLS.loadBLSModule();
        await BLS.bls.init(BLS.bls.BLS12_381);
        BLS.isInitialized = true;
    }
    static guardInitialized() {
        if (!BLS.isInitialized) {
            throw new errors_1.ErrInvariantFailed("BLS modules are not initalized. Make sure that 'await BLS.initIfNecessary()' is called correctly.");
        }
    }
}
exports.BLS = BLS;
BLS.isInitialized = false;
class ValidatorSecretKey {
    constructor(buffer) {
        BLS.guardInitialized();
        assertions_1.guardLength(buffer, exports.VALIDATOR_SECRETKEY_LENGTH);
        this.secretKey = new BLS.bls.SecretKey();
        this.secretKey.setLittleEndian(Uint8Array.from(buffer));
        this.publicKey = this.secretKey.getPublicKey();
    }
    static fromPem(text, index = 0) {
        return pem_1.parseValidatorKey(text, index);
    }
    generatePublicKey() {
        let buffer = Buffer.from(this.publicKey.serialize());
        return new ValidatorPublicKey(buffer);
    }
    sign(message) {
        let signatureObject = this.secretKey.sign(message);
        let signature = Buffer.from(signatureObject.serialize());
        return signature;
    }
    hex() {
        return this.valueOf().toString("hex");
    }
    valueOf() {
        return Buffer.from(this.secretKey.serialize());
    }
}
exports.ValidatorSecretKey = ValidatorSecretKey;
class ValidatorPublicKey {
    constructor(buffer) {
        assertions_1.guardLength(buffer, exports.VALIDATOR_PUBKEY_LENGTH);
        this.buffer = Buffer.from(buffer);
    }
    hex() {
        return this.buffer.toString("hex");
    }
    valueOf() {
        return this.buffer;
    }
}
exports.ValidatorPublicKey = ValidatorPublicKey;
